core/src/main/java/com/linecorp/armeria/server/ServerConfig.java

```diff
@@ -169,10 +178,18 @@
                                    gracefulShutdownQuietPeriod, "gracefulShutdownQuietPeriod");
 
         requireNonNull(blockingTaskExecutor, "blockingTaskExecutor");
-        blockingTaskExecutor =
-                ExecutorServiceMetrics.monitor(meterRegistry, blockingTaskExecutor,
-                                               "blockingTaskExecutor", "armeria");
+        if (!(blockingTaskExecutor instanceof ThreadPoolExecutor ||
+              blockingTaskExecutor instanceof ForkJoinPool)) {
+            logger.warn("Cannot record metrics with {}. (expected: ThreadPoolExecutor or ForkJoinPool)",
+                        blockingTaskExecutor);
+        }
+        ExecutorServiceMetrics.monitor(meterRegistry, ((BlockingTaskExecutor) blockingTaskExecutor).unwrap(),
+                                       "blockingTaskExecutor", "armeria");
```

Comment:
- The WARN log message will always be logged, which is probably not what we really want.
- How about making it a little bit more obvious like this, so we are sure we didn't accidently called the wrong `monitor()` method?
```suggestion
        new ExecutorServiceMetrics(blockingTaskExecutor.unwrap(),
                                   "blockingTaskExecutor", "armeria",
                                   ImmutableList.of()).bindTo(meterRegistry);
```